src/compas_fab/planning/state.py

```diff
@@ -0,0 +1,280 @@
+from compas.data import Data
+from compas.geometry import Frame, Transformation  # noqa F401
+from compas.robots import Configuration  # noqa F401
+
+try:
+    from typing import Optional  # noqa F401
+    from typing import Dict  # noqa F401
+except ImportError:
+    pass
+
+__all__ = [
+    "SceneState",
+    "WorkpieceState",
+    "ToolState",
+    "RobotState",
+]
+
+
+class SceneState(Data):
+    """Container for the states of all workpieces, tools and robot in a static scene.
+
+    Current implementation supports multiple workpieces and tools, but only one robot.
```

Comment:
For the record as we have discussed in our meeting: the scene state will only be able to keep track of one single robot (aka one single robot model / URDF model) . 

For Complex robots like the RFL, the current workflow with a compas_fab model can support multiple planning groups. All the backends can also support multiple planning groups. It will be the role of Actions and planning calls to keep track of which planning group to use.